sql/catalyst/src/main/scala/org/apache/spark/sql/types/StructType.scala

```diff
@@ -526,6 +526,25 @@ object StructType extends AbstractDataType {
       case _ => dt
     }
 
+  /**
+   * Function to merge the two DataTypes that is compatible with the left
+   * and right side. The order is:
+   *
+   * 1. Merge Arrays, where the type of the Arrays should be compatible
+   * 2. Merge Maps, where the type of the Maps should be compatible
+   * 3. Merge Structs, where the struct recursively checked for compatibility
+   * 4. Merge DecimalType, where the scale and precision should be equal
+   * 5. Merge UserDefinedType, where the underlying class is equal
+   * 6. Merge UserDefinedType into DataType, where the underlying DataType is equal
+   * 7. Merge DataType, where we check if the DataTypes are compatible
+   * 8. Unable to determine compatibility or incompatible, throw exception
```

Comment:
Can you merge this comment with ones in L447-455, then move there? I personally think we don't need comments here.

Reply:
Sure